### PR TITLE
Update profile feature complete.

### PR DESCRIPTION
**Pull request by AdrianGonzalez1**
userupdateprofile

### CODE REVIEW

**Comment by AdrianGonzalez1** [comment]
userupdateprofile

File: js/main.js [hunk 1]
```diff
@@ -119,3 +140,21 @@ document.addEventListener('DOMContentLoaded', function () {
     viewSwap('profile');
   }
 });
+
+document.addEventListener('click', function (event) {
+  if (event.target.tagName !== 'A') {
+    return;
+  }
+  if (event.target.tagName === 'A') {
```
Comment (codingcodymiller) [inline js/main.js]: You can remove this conditional, you want the user to be able to go to `profile` if they have user data, and you are already redirecting to `edit-profile` if they don't in the conditional below 👍

File: js/main.js [hunk 2]
```diff
@@ -87,12 +95,19 @@ function renderProfile(data) {
...
   iconCol.appendChild(bioDiv);
+  anchorDiv.appendChild(anchorElement);
+  iconCol.appendChild(anchorDiv);
   imgRow.appendChild(iconCol);
   containerDiv.appendChild(imgRow);
 
   return containerDiv;
 }
+
+// CAN YOU NOT HEAR ME????
```
Comment (codingcodymiller) [inline js/main.js:106]: I hear you!!!!!!

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

File: js/main.js [hunk 3]
```diff
@@ -142,18 +142,10 @@ document.addEventListener('DOMContentLoaded', function () {
...
-  if (event.target.tagName === 'A') {
-    viewSwap('edit-profile');
-  }
   if (event.target.getAttribute('data-view') === 'profile') {
-    // if they do have a username swap to view profile
-    // if they do not have a username swap to edit-profile
     if (data.profile.username !== '') {
       viewSwap('profile');
-    } else if (data.profile.username === '') {
+    } else if (data.profile.username === '' && event.target.tagName === 'A') {
```
Comment (codingcodymiller) [inline js/main.js]: This condition was fine before 👍

File: js/main.js [hunk 4]
```diff
@@ -142,18 +142,10 @@ document.addEventListener('DOMContentLoaded', function () {
 });
 
 document.addEventListener('click', function (event) {
-  if (event.target.tagName !== 'A') {
```
Comment (codingcodymiller) [inline js/main.js]: Oops! you should have kept this conditional, this is our check to see if the user actually clicked an anchor tag 👍

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Pull request merged**